### PR TITLE
PMM-12989 Reduce error logs from diagnostic data on arbiter nodes

### PR DESCRIPTION
**Pull request by idoqo**
[PMM-12989](https://jira.percona.com/browse/PMM-12989) (optional, if ticket reported)

- [ ] https://github.com/Percona-Lab/pmm-submodules/pull/3590
---

Below we provide a basic checklist of things that would make it a good PR:
- Make sure to sign the CLA (Contributor License Agreement).
- Make sure all tests pass.
- Keep current with the target branch and fix conflicts if necessary.
- Update jira ticket description if necessary.
- Attach screenshots and/or console output to the jira ticket to confirm new behavior, if applicable.
- Leave notes to the reviewers if you need to focus their attention on something specific.

Once all checks pass and the code is ready for review, please add `pmm-review-exporters` team as the reviewer. That would assign people from the review team automatically. Report any issues on our [Forum](https://forums.percona.com).

[PMM-12989]: https://perconadev.atlassian.net/browse/PMM-12989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -91,18 +91,24 @@ It will install `goimports`, `goreleaser`, `golangci-lint` and `reviewdog`.
 
 The testing sandbox starts `n` MongoDB instances as follows:
 
-- 3 Instances for shard 1 at ports 17001, 17002, 17003
-- 3 instances for shard 2 at ports 17004, 17005, 17006
+- 3 Instances for shard 1 at ports 17001, 17002, 17003 (with no authentication)
+- 3 instances for shard 2 at ports 17004, 17005, 17006 (with authentication enabled)
```
Comment (JiriCtvrtka) [inline CONTRIBUTING.md:95]: 👍

**Review by JiriCtvrtka — APPROVED** [review]

File: exporter/diagnostic_data_collector.go [hunk 2]
```diff
@@ -66,39 +66,46 @@ func (d *diagnosticDataCollector) collect(ch chan<- prometheus.Metric) {
...
-		metrics = append(metrics, securityMetric)
+		securityMetric, err := d.getSecurityMetricFromLineOptions(client)
+		if err != nil {
+			logger.Errorf("failed to run command: getCmdLineOptions: %s", err)
+		} else if securityMetric != nil {
+			metrics = append(metrics, securityMetric)
+		}
 	}
 
 	if d.compatibleMode {
```
Comment (BupycHuk) [inline exporter/diagnostic_data_collector.go:111]: at this point `m` might be empty, so it makes not much sense to call lines https://github.com/percona/mongodb_exporter/pull/820/files#diff-f08bb8e2230b960e7f7941dbdf020e3aa607391de1dc0f8683c315265d77b2f6R112-R116
I would include them to else part of the statement above

Comment (BupycHuk) [reply exporter/diagnostic_data_collector.go:111]: BTW can we get diagnostic data from mongos?

File: exporter/v1_compatibility.go [hunk 3]
```diff
@@ -815,9 +815,7 @@ func specialMetrics(ctx context.Context, client *mongo.Client, m bson.M, l *logr
 	}
 
 	if nodeType == typeMongod || nodeType == typeArbiter {
-		if engine, err := storageEngine(m); err != nil {
-			l.Errorf("cannot retrieve engine type: %s", err)
-		} else {
```
Comment (BupycHuk) [inline exporter/v1_compatibility.go:820]: do we know when we can't retrieve engine? it's important to not just hide an error, but hide it only for reason.

File: exporter/diagnostic_data_collector.go (hunk 2 above)
Comment (idoqo) [reply exporter/diagnostic_data_collector.go:111]: > BTW can we get diagnostic data from mongos?

No actually, the command won't fail, but `m["data"]` will be empty.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**